test/test_jit.py

```diff
@@ -1755,14 +1771,124 @@ def forward(self, x):
                 x = self.bn2(x)
                 x = self.relu(x)
                 return x
-        for bias in [True, False]:
-            eager = TestModule(bias).eval()
-            scripted = torch.jit.script(eager).copy()
-            torch._C._jit_pass_dedup_module_uses(scripted._c)
-
-            folded = wrap_cpp_module(torch._C._jit_pass_fold_convbn(scripted._c))
-            x = torch.rand(1, 5, 6, 6)
-            self.assertEqual(eager(x), scripted(x))
+
+        for tracing_mode in [True, False]:
+            for bias in [True, False]:
+                eager = TestModule(bias).eval()
+                if tracing_mode:
+                    x = torch.rand(1, 5, 6, 6)
+                    scripted_or_traced = torch.jit.trace(eager, x).copy()
+                else:
+                    scripted_or_traced = torch.jit.script(eager).copy()
+                torch._C._jit_pass_dedup_module_uses(scripted_or_traced ._c)
+                folded = wrap_cpp_module(torch._C._jit_pass_fold_convbn(scripted_or_traced ._c))
+                x = torch.rand(1, 5, 6, 6)
+                self.assertEqual(eager(x), scripted_or_traced (x))
+
+    def test_foldbn_complex_cases(self):
+        # Test that we find Conv-BN patterns in submodules
```

Comment:
can you add more decription for this one? looks like this tests different possible combinations of `bias=True/False` `affine=True/False`

Reply:
Yes that is correct. I will add the description.